packages/xo-server/src/xapi/mixins/patching.mjs

```diff
@@ -651,4 +637,22 @@ export default {
       throw error
     }
   },
+
+  @decorateWith(deferrable)
+  async rollingPoolUpdate($defer, { xsCredentials } = {}) {
```

Comment:
`hasMissingPatchesByHost` should be calculated only on `rollingPoolUpdate`, otherwise hosts will not restart in some cases.